lualib/lua_selectors/transforms.lua

```diff
@@ -36,7 +37,7 @@ local transform_function = {
     },
     ['map_type'] = 'string',
     ['process'] = function(inp, _)
-      return inp:lower(),'string'
```

Comment:
I would suggest to add an additional selector instead of introducing a slashing change. For the most of the cases `lower` should be treated as `ascii_lower` for performance and compatibility reasons. If we want a `utf8` version of lower, it is good to have a `lower_utf8` selector. Instead of transforming all to text, I would suggest to use the following code:

```lua
function(inp, ty)
  return rspamd_util.lower_utf8(inp),ty
end
```

It will deal with both native strings and texts without implicit copy and type transformation.

Reply:
I see, that makes sense. I pushed another version with lower_utf8 selector